### PR TITLE
[WIP] Convert object dtype

### PR DESCRIPTION
**Pull request by andrewgsavage**
Addresses https://github.com/hgrecco/pint-pandas/issues/55

### CODE REVIEW

**Comment by andrewgsavage** [comment]
readu for review

File: CHANGES [hunk 1]
```diff
@@ -6,6 +6,7 @@ pint-pandas Changelog
 
 - Support for <NA> values in columns with integer magnitudes 
 - Support for magnitudes of any type, such as complex128 or tuples #146
+- A DataFrame/Series.pint.convert_object_dtype() function has been added to create PintArrays from Series of quantitys.
```
Comment (MichaelTiemannOSC) [inline CHANGES:9]: typo: quantitys to quantities

File: pint_pandas/pint_array.py [hunk 2]
```diff
@@ -933,6 +950,19 @@ def _validate(obj):
                 "dtype '{}'.".format(obj.dtype)
             )
 
+    @staticmethod
+    def _is_object_dtype_and_quantity(obj):
+        return obj.dtype == "object" and all(
+            [(isinstance(item, _Quantity) or pd.isna(item)) for item in obj.values]
```
Comment (MichaelTiemannOSC) [inline pint_pandas/pint_array.py:956]: Does this play well with uncertainties?

**Pull request merged**